homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -466,6 +466,7 @@ class LeafEntity(Entity):
     def __init__(self, car):
         """Store LeafDataStore upon init."""
         self.car = car
+        self._attr_unique_id = f"{self.car.leaf.vin}_{self.__class__.__name__}".lower()
```

Comment:
I would not recommend using the class name in the unique ID.
While a nice shortcut, it makes refactoring this integration to EntityDescriptions in the future way harder.

Reply:
Hi @frenck, Thanks for the review.  I had trouble working out the EntityDescriptions requirements (seems to be hard to find docs on this - I saw it mentioned in the [State of the Open Home at 57:20](https://youtu.be/6ZMXE5PXPqU?t=3440) but not much else).  I've taken the naive approach in the interim.  Open to better suggestions! :)

Reply:
You don't have to use them at this point, however, unique IDs should be based on data, not code structure. Simply, because changing structure in the future (e.g., when using EntityDescription or something else... doesn't matter), the unique ID changes because structure changes.

Reply:
I've updated the code to explicitly set a unique id for each entity